### PR TITLE
Do not allow to return to current project if it is a deleted cloud project

### PR DESCRIPTION
**Pull request by suricactus**
1) download cloud project
2) open it
3) go back to qfc projects screen
4) delete it
5) go to main screen
6) clicking top-right back to current project shows some non-sense

### CODE REVIEW

**Comment by duke-nyuki** [comment]
Task linked: [QF-3059 Fix back button on welcome screen when no project is opened](https://app.clickup.com/t/2192114/QF-3059)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 4a3cc15fe57ebf247077040a94b567cf9ffc0ad5: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4958-e809ddeaa64c6d06cc166fbd8520188e6dd76048-android-arm64.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4958-e809ddeaa64c6d06cc166fbd8520188e6dd76048-android-arm-neon.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4958-e809ddeaa64c6d06cc166fbd8520188e6dd76048-android-x64.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4958-e809ddeaa64c6d06cc166fbd8520188e6dd76048-android-x86.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4958-e809ddeaa64c6d06cc166fbd8520188e6dd76048-android-arm64.apk)
- [all access arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4958-e809ddeaa64c6d06cc166fbd8520188e6dd76048-android-arm-neon.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4958-e809ddeaa64c6d06cc166fbd8520188e6dd76048-android-x64.apk)
- [all access x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4958-e809ddeaa64c6d06cc166fbd8520188e6dd76048-android-x86.apk)
</details>

File: src/qml/QFieldCloudScreen.qml [hunk 1]
```diff
@@ -554,6 +554,10 @@ Page {
             cloudProjectsModel.removeLocalProject(projectActions.projectId)
             iface.removeRecentProject(projectActions.projectLocalPath);
             welcomeScreen.model.reloadModel()
+
+            if (projectActions.projectLocalPath === qgisProject.fileName) {
+              qgisProject.fileName = ""
```
Comment (nirvn) [inline src/qml/QFieldCloudScreen.qml:559]: This is a bit of a hack here, shouldn't we instead properly close the project?

Settings the fileName to "" edits the project, it doesn't clear it of its content.

Comment (suricactus) [reply src/qml/QFieldCloudScreen.qml:559]: `QgsProject.clear()` is not invocable, so it needs a utility function. Will close this PR and come back to this when I have time.

Comment (nirvn) [reply src/qml/QFieldCloudScreen.qml:559]: @suricactus , yeah, I was thinking you could add a iface.clearProject() , that'd be easy enough

Comment (suricactus) [reply src/qml/QFieldCloudScreen.qml:559]: Nice reminder that we have this module. Seems I was not the only one to thing where to put some utility function :) .

File: src/core/appinterface.cpp [hunk 2]
```diff
@@ -191,6 +192,11 @@ void AppInterface::closeSentry() const
 #endif
 }
 
+void AppInterface::clearProject() const
+{
+  QgsProject::instance()->clear();
```
Comment (nirvn) [inline src/core/appinterface.cpp:197]: This is not ideal, I think it'd be better to add a QgisMobileApp:clearProject function in which we mProject->clear(), and call that in AppInterface (i.e. mApp->clearProject()).

I know we refer to QgsProject::instance() in multiple places :) but I feel when it's that easy and uncomplicated to do, we should avoid.

Comment (nirvn) [reply src/core/appinterface.cpp:197]: Otherwise, super happy with this!

**Review by nirvn — APPROVED** [review]

**Pull request merged**